bitnami/tensorflow-resnet/values.yaml

```diff
@@ -86,44 +87,11 @@ readinessProbe:
   failureThreshold: 6
   successThreshold: 1
 
-## Prometheus Exporter / Metrics
+## Add the corresponding annotations so Prometheus can collect Tensorflow Serving metrics
 ##
 metrics:
   enabled: false
-  image:
-    registry: docker.io
-    repository: ynqa/tensorflow-serving-exporter
-    tag: latest
-    pullPolicy: IfNotPresent
-    ## Optionally specify an array of imagePullSecrets.
-    ## Secrets must be manually created in the namespace.
-    ## ref: https://kubernetes.io/docs/tasks/configure-pod-container/pull-image-private-registry/
-    ##
-    # pullSecrets:
-    #   - myRegistryKeySecretName
-  port: 9118
-  ## Configure extra options for liveness and readiness probes
-  ## ref: https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-probes/#configure-probes)
-  livenessProbe:
-    enabled: true
-    initialDelaySeconds: 15
-    periodSeconds: 10
-    timeoutSeconds: 5
-    failureThreshold: 2
-    successThreshold: 1
-
-  readinessProbe:
-    enabled: true
-    initialDelaySeconds: 5
-    periodSeconds: 10
-    timeoutSeconds: 1
-    failureThreshold: 6
-    successThreshold: 1
-  ## Metrics exporter pod Annotation and Labels
   podAnnotations:
     prometheus.io/scrape: "true"
-    prometheus.io/port: "9118"
-    ## Metrics exporter resource requests and limits
-    ## ref: http://kubernetes.io/docs/user-guide/compute-resources/
-    ##
-  # resources: {}
+    prometheus.io/path: "/monitoring/prometheus/metrics"
+    prometheus.io/port: "8501"
```

Comment:
Correct me if I'm wrong but we're setting the Prometheus annotations in the Tensorflow pods, aren't we? 

I wonder if it makes sense to create a specific service `metrics-svc.yaml` with the annotations and redirecting traffic to the restApiPort. If we do so, and we add the annotations to this service, we don't nee to add the annotations to the pods.

Reply:
I not sure about it. The port `8501` exposes the Rest API of Tensorflow serving. Among other things, it includes metrics for Prometheus in the `/monitoring/prometheus/metrics`, but it also has other endpoints not related to metrics, i.e. `/models/resnet`.

Reply:
🤔 in this case, I'd leave it as it's. Thanks for the explanation.